src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -140,17 +162,21 @@ public static PhysicalAddress Parse(string address)
             {
                 int value = (int)address[i];
 
-                if (value >= 0x30 && value <= 0x39)
+                if (value >= (int)'0' && value <= (int)'9')
                 {
                     value -= 0x30;
```

Comment:
Since you're changing the values in the ifs, it'd be good to change these, too, e.g. `value -= (int)'0'`